src/qix.js

```diff
@@ -5,12 +5,75 @@ import SuspendResume from './suspend-resume';
 import Intercept from './intercept';
 import ApiCache from './api-cache';
 
+/**
+ * The enigma.js configuration object.
+ * @interface Configuration
+ * @property {Object} schema Object containing the specification for the API to generate.
+ * Corresponds to a specific version of the QIX Engine API.
+ * @property {String} url String containing a proper websocker URL to QIX Engine.
+ * @property {Function} [createSocket] A function to use when instantiating the WebSocket,
+ * mandatory for Node.js.
+ * @property {Object} [Promise] ES6-compatible Promise library.
+ * @property {Boolean} [suspendOnClose=false] Set to true if the session should be suspended
+ * instead of closed when the websocket is closed.
+ * @property {Array<Mixin>} [mixins=[]] Mixins to extend/augment the QIX Engine API. Mixins
+ * are applied in the array order.
+ * @property {Array} [requestInterceptors=[]] Interceptors for augmenting requests before they
+ * are sent to QIX Engine. Interceptors are applied in the array order.
+ * @property {Array} [responseInterceptors=[]] Interceptors for augmenting responses before they
+ * are passed into mixins and end-users. Interceptors are applied in the array order.
+ * @property {Object} [protocol={}] An object containing additional JSON-RPC request parameters.
+ * @property {Boolean} [protocol.delta=true] Set to false to disable the use of the
+ * bandwidth-reducing delta protocol.
+ */
+
+/**
+ * Mixin object to extend/augment the QIX Engine API
+ * @interface Mixin
+ * @property {String|Array<String>} types String or array of strings containing the API-types that
+ * will be mixed in.
+ * @param {Object} [extend] Object literal containing the methods that will be extended on the
```

Comment:
I think all of these should be `@property` instead of `@param`

Reply:
Yes! My bad..